### ISSUE TITLE
nomatch = NULL does not work for factors

### ISSUE THREAD

**Issue by jhuovari** [vincentarelbundock/countrycode#192]
Using `nomatch = NULL`  for a string gives an original value, but a NA for a factor. 

```
countrycode(c('USA', 'DZA', 'EU'), 'iso3c', 'country.name', nomatch = NULL)
## [1] "United States" "Algeria"       "EU"  

countrycode(factor(c('USA', 'DZA', 'EU')), 'iso3c', 'country.name', nomatch = NULL)
## [1] "United States" "Algeria"       NA             
## Warning messages:
## 1: In countrycode(factor(c("USA", "DZA", "EU")), "iso3c", "country.name",  :
##  The origin and destination codes are not of the same
##                   class. Filling-in bad matches with NA instead.
## 2: In countrycode(factor(c("USA", "DZA", "EU")), "iso3c", "country.name",  :
##   Some values were not matched unambiguously: EU

```

**Comment by cjyetman** [comment]
Thanks! As far as I know, that is unintended behavior, though the function does identify the problem and warns about it.

for reference, this ultimately comes down to this line:
https://github.com/vincentarelbundock/countrycode/blob/master/R/countrycode.R#L206
`sane_sourcevar <- class(sourcevar)[1] == class(destination_vector)[1]`

Hopefully one of us will have time to look at this soon. (of course, ideas or fixes from anyone else are also very welcome)

**Comment by cjyetman** [comment]
this is fixed in c2042d8750a25da72ed9d98ba09e141efccd9d2d
